pkg/cmd/historical_logs.go

```diff
@@ -107,40 +127,321 @@ func (o *LogParameters) Execute(streams genericclioptions.IOStreams) error {
+			response, err := o.makeHttpRequest(ApiUrl)
+
+			if err != nil {
+				return err
+			}
+
+			responseBody, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response: %v", err)
+			}
+
+			err = getLogs(responseBody, &logList, o.Prefix)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+
+	} else {
+
+		response, err := o.makeHttpRequest(ApiUrl)
+		if err != nil {
+			return err
+		}
+		responseBody, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response: %v", err)
+		}
+
+		err = getLogs(responseBody, &logList, o.Prefix)
+		if err != nil {
+			return err
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *LogParameters) makeHttpRequest(baseUrl string) (*http.Response, error) {
+
+	var urlBuilder strings.Builder
+	urlBuilder.WriteString(baseUrl)
+	numParameters := 0
+	if len(o.Podname) > 0 {
+		urlBuilder.WriteString("?")
+		urlBuilder.WriteString("podname=" + o.Podname)
+		numParameters = numParameters + 1
+	}
+	if len(o.Namespace) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("namespace=" + o.Namespace)
+		numParameters = numParameters + 1
+	}
+	if len(o.Limit) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("maxlogs=" + o.Limit)
+		numParameters = numParameters + 1
+	}
+	if len(o.Level) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("level=" + o.Level)
+		numParameters = numParameters + 1
+	}
+	if len(o.StartTime) > 0 && len(o.EndTime) > 0 {
+		if numParameters == 0 {
+			urlBuilder.WriteString("?")
+		} else {
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("starttime=" + o.StartTime)
+		urlBuilder.WriteString("&finishtime=" + o.EndTime)
+		numParameters = numParameters + 1
+	}
+	response, err := http.Get(urlBuilder.String())
+
 	if err != nil {
-		return fmt.Errorf("Failed to read Response: %v", err)
+		return nil, fmt.Errorf("http request failed: %v", err)
 	}
+	return response, nil
+}
+
+func printLogs(logList []string, streams genericclioptions.IOStreams, limit string) error {
+
+	if len(logList) == 0 {
+		_, err := fmt.Fprintf(streams.Out, "No Logs Present, or logs from invalid resources were requested")
+		if err != nil {
+			return fmt.Errorf("an error occurred while printing logs: %v", err)
+		}
+		return nil
+	}
+
+	var maxLogs int
+	var err error
+
+	if len(limit) > 0 {
+		maxLogs, err = strconv.Atoi(limit)
+		if err != nil {
+			return fmt.Errorf("incorrect \"limit\" value entered")
+		}
+	}
+
+	for logCount, log := range logList {
+
+		if len(limit) > 0 && logCount > maxLogs {
```

Comment:
And for `maxlogs=0` it is printing 1 log. I believe this should be changed to `if len(limit) > 0 && logCount >= maxLogs`. I'll check this again and confirm.

Reply:
This needs to be changed. Since index starts from 0, it always needs to print logs till `maxLogs` - 1, and not till `maxLogs`.